src/journal/journald-syslog.c

```diff
@@ -397,24 +406,33 @@ void server_process_syslog_message(
         if (syslog_ts_len > 0) {
                 const size_t hlen = strlen("SYSLOG_TIMESTAMP=");
 
-                t = newa(char, hlen + raw_len);
+                t = newa(char, hlen + syslog_ts_len);
                 memcpy(t, "SYSLOG_TIMESTAMP=", hlen);
                 memcpy(t + hlen, syslog_ts, syslog_ts_len);
 
                 iovec[n++] = IOVEC_MAKE(t, hlen + syslog_ts_len);
         }
 
-        a = strjoina("MESSAGE=", msg);
+        msg_msg = strjoin("MESSAGE=", msg);
+        if (!msg_msg) {
+                log_oom();
+                return;
+        }
         iovec[n++] = IOVEC_MAKE_STRING(a);
```

Comment:
It seems that `msg_msg` should be here instead of `a`. Otherwise, it will crash with something like
```
=================================================================
==14==ERROR: AddressSanitizer: SEGV on unknown address 0x02007f528a5d (pc 0x000000485abe bp 0x7fffd4c29450 sp 0x7fffd4c28be0 T0)
==14==The signal is caused by a READ memory access.
    #0 0x485abd in AddressIsPoisoned /src/llvm/projects/compiler-rt/lib/asan/asan_mapping.h:387
    #1 0x485abd in QuickCheckForUnpoisonedRegion /src/llvm/projects/compiler-rt/lib/asan/asan_interceptors_memintrinsics.h:32
    #2 0x485abd in __interceptor_strlen.part.33 /src/llvm/projects/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:356
    #3 0x554596 in server_process_syslog_message /work/build/../../src/systemd/src/journal/journald-syslog.c:421:22
    #4 0x534242 in LLVMFuzzerTestOneInput /work/build/../../src/systemd/src/fuzz/fuzz-journald.c:28:9
    #5 0x59c9e2 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/libfuzzer/FuzzerLoop.cpp:580:15
    #6 0x574d0a in fuzzer::RunOneTest(fuzzer::Fuzzer*, char const*, unsigned long) /src/libfuzzer/FuzzerDriver.cpp:280:6
    #7 0x5803d5 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/libfuzzer/FuzzerDriver.cpp:715:9
    #8 0x564c2c in main /src/libfuzzer/FuzzerMain.cpp:20:10
    #9 0x7f6c5661782f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #10 0x420e38 in _start (/out/fuzz-journald+0x420e38)

```

Reply:
Ouch!

Reply:
Fixed.